default.vcl.tmpl

```diff
@@ -220,6 +220,27 @@ sub vcl_backend_response {
     set beresp.do_gzip = true;
   }
 
+  # if there is no s-maxage header, set a default cache time
+  # 59 it's because we can differ between the default header and header's set in the application
+  if (beresp.http.Cache-Control !~ "s-maxage") {
+      set beresp.ttl = 59s;
```

Comment:
How is that much different to https://github.com/livingdocsIO/dockerfile-varnish/blob/master/default.vcl.tmpl#L193 ?

Can we adjust the config in there?

Reply:
It differs only in the condition. We need the s-maxage set explicitly because if not present in the response headers our App is taking the value of max-age by default.

Reply:
Ok, I'd put all configs next to each other. Let's have a call if necessary.

Reply:
@marcbachmann this snippet was necessary for Varnish 5 because, when `s-max-age` and `max-age` were present, it would take `max-age` as the TTL. Varnish 4 was taking `s-max-age`. 

The app's offline capability relies on a `max-age` being e..g 3 days (so it stays on the phone longer). At the same time we want varnish to have relatively fresh copies, se we set `s-max-age` to 1 min. 

The apps will add a `Cache-Control: max-age=0` in their requests whenever they do a "refresh" of the content, thus forcing the on-device cache to revalidate its contents.

Reply:
> @marcbachmann this snippet was necessary for Varnish 5 because, when s-max-age and max-age were present, it would take max-age as the TTL. Varnish 4 was taking s-max-age.

Now you're confusing me. There's no such thing as `s-max-age`. There's only `s-maxage`. So the behavior you've mentioned would be correct.

The rest you've mentioned makes sense, I was just questioning the hardcoded value and not reusing the other code as it's basically for almost the same use case.

Reply:
I meant `s-maxage`, correct. @ile2807 will remove line 225, 226, you are right that they don't make sense.